sycl/plugins/level_zero/pi_level0.cpp

```diff
@@ -363,6 +363,38 @@ pi_result _pi_device::initialize() {
   ZeDeviceComputeProperties.version =
       ZE_DEVICE_COMPUTE_PROPERTIES_VERSION_CURRENT;
   ZE_CALL(zeDeviceGetComputeProperties(ZeDevice, &ZeDeviceComputeProperties));
+
+  ZeAvailMemCount = 0;
+  ZE_CALL(zeDeviceGetMemoryProperties(ZeDevice, &ZeAvailMemCount, nullptr));
+  // Confirm at least one memory is available in the device
+  assert(ZeAvailMemCount > 0);
+
+  try {
+    ZeDeviceMemoryProperties =
+        new ze_device_memory_properties_t[ZeAvailMemCount]();
```

Comment:
This needs to be deleted in piDeviceRelease

Reply:
Missed that that was not being done. Adding it.

Reply:
@jandres742 , did you add it?